### ISSUE TITLE
Un-braced nested if/else gives false positive for unreachable_code_linter

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2710]
Found in {bit64}:

```r
lint(linters = unreachable_code_linter(), '
function(x, ...){
    if(is.null(env))
        has.cache <- FALSE
    else if (exists("nunique", envir=env, inherits=FALSE))
        return(get("nunique", envir=env, inherits=FALSE))
    else
        has.cache <- TRUE
}
')
# <text>:7:5: warning: [unreachable_code_linter] Remove code and comments coming after return() or stop().
#     else
#     ^~~~
```

https://github.com/r-lib/bit64/blob/cb4552b8345550d158cc7ae9b937b009f4dcd200/R/cache.R#L358-L363

**Comment by MichaelChirico** [comment]
Fixed by #2900:

```
ℹ No lints found.
```
